h/config.py

```diff
@@ -30,105 +27,91 @@
                 b"\x9f\x1b\x96\xc1\xfa\x8b\x19\x82\xa3[\x19\xcb\xa4\x1a\x0f"
                 b"\xe4\xcb\r\x17\x7f\xfbh\xd5^W\xdb\xe6")
 
-# The list of all settings read from the system environment. These are in
-# reverse-priority order, meaning that later settings trump earlier settings.
-SETTINGS = [
-    # Mailer configuration for Mandrill
-    mandrill_settings,
 
+def configure(environ=None, settings=None):
+    environ = environ or os.environ
+    settings = settings or {}
+    settings_manager = SettingsManager(settings, environ)
     # Configuration for external components
-    EnvSetting('broker_url', 'BROKER_URL'),
-    EnvSetting('es.client_poolsize', 'ELASTICSEARCH_CLIENT_POOLSIZE',
-               type=int),
-    EnvSetting('es.client.max_retries', 'ELASTICSEARCH_CLIENT_MAX_RETRIES', type=int),
-    EnvSetting('es.client.retry_on_timeout', 'ELASTICSEARCH_CLIENT_RETRY_ON_TIMEOUT', type=asbool),
-    EnvSetting('es.client.timeout', 'ELASTICSEARCH_CLIENT_TIMEOUT', type=float),
-    EnvSetting('es.host', 'ELASTICSEARCH_HOST'),
-    EnvSetting('es.index', 'ELASTICSEARCH_INDEX'),
-    EnvSetting('es.aws.access_key_id', 'ELASTICSEARCH_AWS_ACCESS_KEY_ID'),
-    EnvSetting('es.aws.region', 'ELASTICSEARCH_AWS_REGION'),
-    EnvSetting('es.aws.secret_access_key', 'ELASTICSEARCH_AWS_SECRET_ACCESS_KEY'),
-    EnvSetting('mail.default_sender', 'MAIL_DEFAULT_SENDER'),
-    EnvSetting('mail.host', 'MAIL_HOST'),
-    EnvSetting('mail.port', 'MAIL_PORT', type=int),
-    EnvSetting('sqlalchemy.url', 'DATABASE_URL', type=database_url),
-    EnvSetting('statsd.host', 'STATSD_HOST'),
-    EnvSetting('statsd.port', 'STATSD_PORT', type=int),
-    EnvSetting('statsd.prefix', 'STATSD_PREFIX'),
+    settings_manager.set('broker_url', 'BROKER_URL')
+    settings_manager.set('es.client_poolsize', 'ELASTICSEARCH_CLIENT_POOLSIZE',
+                         type=int)
```

Comment:
I think you renamed `type` to `type_` but are still using `type` in `config.py` (if I run `make dev` it crashes)